frontend/README.adoc

```diff
@@ -61,41 +62,187 @@ The Kiali UI is a React application written in Typescript.
+
+- **React class:**
+
+Use `withTranslation` high order component. More info link:https://react.i18next.com/latest/withtranslation-hoc[here]
+
+You have to embed React class with `withTranslation` function (similar to redux connect function):
+
+[source, typescript]
+----
+export const Menu = withTranslation(I18N_NAMESPACE)(MenuComponent);
+----
+
+Besides, `WithTranslation` props must be added to React class properties to include translation `t` function (like Redux properties)
+
+[source, typescript]
+----
+type MenuProps = WithTranslation & {
+   ...
+};
+----
+
+Then you will be able to use `t` function as any other React class property to translate any label (variable or string literal).
+[source, typescript]
+----
+<h1>{this.props.t(title)}</h1>
+----
+
+
+- **React hook:**
+
+Use useTranslation hook to get translation `t` function. More info link:https://react.i18next.com/latest/usetranslation-hook[here]. 
+
+Any label (variable or string literal) can be translated with `t` function within the component.
+
+[source, typescript]
 ----
-$ curl -u admin:admin http://localhost:3000/api
-Namespaces: [default istio-system kube-public kube-system myproject openshift openshift-infra openshift-node]
+const { t } = useTranslation(I18N_NAMESPACE);
+
+<h1>{t(title)}</h1>
+----
+
+- **External variable:**
+
+String literals that are stored in a variable defined outside of React component cannot be translated directly. Instead the translation has to be done when the variable is used within the React component (class or hook).
+
+Problem here is that i18n parser can't extract labels from variables, only from literals. To indicate i18n parser to add that kind of string literals to locale file use `ì18n.t` function. Note that `i18n.t` does not perform any translation, it returns the same string literal on runtime (only for development purposes).
+
+[source, typescript]
+----
+import { i18n } from 'i18n';
+
+const overviewTypes = {
+  app: i18n.t('Apps'),
+  workload: i18n.t('Workloads'),
+  service: i18n.t('Services')
+};
+
+
+const Example: React.FC<Props> = (props: Props) => {
+  const { t } = useTranslation(I18N_NAMESPACE);
+
+  return (
+    <span>{t(overviewTypes.app)}</span>
+    ...
+  )
+}
+----
+
+- **External function:**
 
-Namespace: default Services [docker-registry kubernetes router]
+External functions declared outside of React components cannot use `t` function directly. Instead it must be passed as input parameter (`TFunction` type must be imported from `react-i18next` library, it will fail if imported from `i18next` library):
```

Comment:
```suggestion
External functions declared outside of React components cannot use the `t` function directly. Instead, it must be passed as an input parameter (`TFunction` type must be imported from the `react-i18next` library, it will fail if imported from `i18next` library):
```